docs/layouts/shortcodes/generated/history_server_configuration.html

```diff
@@ -30,7 +30,13 @@
             <td><h5>historyserver.archive.retained-jobs</h5></td>
             <td style="word-wrap: break-word;">-1</td>
             <td>Integer</td>
-            <td>The maximum number of jobs to retain in each archive directory defined by `historyserver.archive.fs.dir`. If set to `-1`(default), there is no limit to the number of archives. If set to `0` or less than `-1` HistoryServer will throw an <code class="highlighter-rouge">IllegalConfigurationException</code>. </td>
+            <td>The maximum number of jobs to retain in each archive directory defined by `historyserver.archive.fs.dir`. When this option is enabled together with the `historyserver.archive.retained-ttl` option, the job archive will be removed if its TTL has expired or the retained job count has been reached. If set to `-1`(default), there is no limit to the number of archives. If set to `0` or less than `-1` HistoryServer will throw an <code class="highlighter-rouge">IllegalConfigurationException</code>. Note, when there are multiple history server instances, please enable the configuration option like following: <ul><li>Enable this feature in only one HistoryServer instance to avoid errors caused by multiple instances simultaneously cleaning up remote files, </li><li>Or you can keep the value of this configuration consistent across them. </li></ul></td>
+        </tr>
+        <tr>
+            <td><h5>historyserver.archive.retained-ttl</h5></td>
+            <td style="word-wrap: break-word;">(none)</td>
+            <td>Duration</td>
+            <td>The time-to-live duration to retain the jobs archived in each archive directory defined by `historyserver.archive.fs.dir`. When this option is enabled together with the `historyserver.archive.retained-jobs` option, the job archive will be removed if its TTL has expired or the retained job count has been reached. Note, when there are multiple history server instances, please enable the configuration option like following: <ul><li>Enable this feature in only one HistoryServer instance to avoid errors caused by multiple instances simultaneously cleaning up remote files, </li><li>Or you can keep the value of this configuration consistent across them. </li></ul></td>
```

Comment:
I am not sure what enabled means here. Is this the same as a specified. I assume the default is null which is infinite ttl, I suggest we document this